examples/secrets/values-secret.v2.yaml

```diff
@@ -0,0 +1,81 @@
+# NEVER COMMIT THESE VALUES TO GIT (unless your file only uses generated
+# passwords or only points to files)
+
+# Needed to specify the new format (missing version means old version: 1.0 by default)
+version: 2.0
+
+# These are the vault policies to be created in the vault
+# these are used when we let the vault generate the passwords
+# by setting the 'onMissingValue' attribute to 'generate'
+# See https://developer.hashicorp.com/vault/docs/concepts/password-policies
+vaultPolicies:
+  basicPolicy: |
+    length=10
+    rule "charset" { charset = "abcdefghijklmnopqrstuvwxyz" min-chars = 1 }
+    rule "charset" { charset = "ABCDEFGHIJKLMNOPQRSTUVWXYZ" min-chars = 1 }
+    rule "charset" { charset = "0123456789" min-chars = 1 }
+
+  advancedPolicy: |
+    length=20
+    rule "charset" { charset = "abcdefghijklmnopqrstuvwxyz" min-chars = 1 }
+    rule "charset" { charset = "ABCDEFGHIJKLMNOPQRSTUVWXYZ" min-chars = 1 }
+    rule "charset" { charset = "0123456789" min-chars = 1 }
+    rule "charset" { charset = "!@#$%^&*" min-chars = 1 }
+
```

Comment:
I wonder if it would be good idea to have a few default (named) policies in the playbook and support:

```
enableDefaultVaultPolicies: 
- basic
- advanced
``` 

to avoid a) having this giant block everywhere, and b) keeping it synchronised if it ever needs to be updated

Reply:
I like this idea. Maybe I can just hard-code a reasonable single validatedPatternDefaultPolicy (complex name chosen so it won't overlap ever with something existing) and we push that all the time and examples are guaranteed that they reference it?

Reply:
Ok I added a new hard-coded policy https://github.com/hybrid-cloud-patterns/common/pull/218/commits/36d06e1fccd43f52a57d4d8c56aafa25b4de6ec1#diff-600beb9d1915faccc894389dbc01346727fa651ba20c4212128dca0857351a90R27
I think it is okay to hard-code it as it does nothing if not used. I'd stick to just one as I suspect it will be mainly used for demos and whatnot.